### PR TITLE
sandbox: refactor the sandbox init process

### PR DESCRIPTION
**Pull request by lifupan**
Inorder to support sandbox api, intorduce the sandbox_config struct and split the sandbox start stage from init process.

### CODE REVIEW

File: src/runtime-rs/crates/resource/src/cpu_mem/initial_size.rs
Comment (pmores) [inline src/runtime-rs/crates/resource/src/cpu_mem/initial_size.rs:151]: Do we want to override the config file anytime the `sandbox-memory` annotation is present?

I can see how it could make sense in the context of Sandbox API, just making sure this is a conscious decision.

Also, if so, could you update the reasoning in the comment above the changed line as well?

Comment (lifupan) [reply src/runtime-rs/crates/resource/src/cpu_mem/initial_size.rs]: @pmores   Good catch, actually it's was changed by accident. I had revert this change.

**Review by Apokleos — APPROVED** [review]
Thx @lifupan LGTM

**Review by justxuewei — APPROVED** [review]

**Comment by gkurz** [comment]
>     sandbox: refactor the sandbox init process
>     
>     Inorder to support sandbox api, intorduce the sandbox_config
>     struct and split the sandbox start stage from init process.
>     
>     Signed-off-by: Fupan Li <fupan.lfp@antgroup.com>
> 
>  src/runtime-rs/crates/resource/src/cpu_mem/initial_size.rs   |  51 ++++++++++++++++++++++++++++++++++--------------
>  src/runtime-rs/crates/runtimes/common/src/runtime_handler.rs |   3 ++-
>  src/runtime-rs/crates/runtimes/common/src/sandbox.rs         |  10 +---------
>  src/runtime-rs/crates/runtimes/common/src/types/mod.rs       |  19 +++++++++++++++++-
>  src/runtime-rs/crates/runtimes/linux_container/src/lib.rs    |   3 ++-
>  src/runtime-rs/crates/runtimes/src/manager.rs                | 179 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-------------------------------------------------------------------------
>  src/runtime-rs/crates/runtimes/virt_container/src/lib.rs     |   4 +++-
>  src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs |  73 ++++++++++++++++++++++++++++++++++++++-------------------------------
>  src/runtime-rs/crates/runtimes/wasm_container/src/lib.rs     |   3 ++-
>  9 files changed, 208 insertions(+), 137 deletions(-)
> 

A monolithic patch with such a huge diff stat and a one liner commit message ? How come it that it gets merged in less than 24 hours without even waiting for @pmores to ack the changes matched his on-going review ?

I am very concerned as it clearly doesn't display an exemplary behavior for contributors... :confused: 

@lifupan @justxuewei

**Comment by lifupan** [comment]
Hi @gkurz 

I understand your concerns, and as a community contributor, I also support your carefulness. Although this PR looks like a large diff, it is mainly code refactoring and location movement. Sorry, the commit message are indeed relatively brief. If you think it is necessary to modify the commit messages to be more detailed, I can revert this merge and add some detail info.

In addition, for this PR, I just want to make it easier for the reviewer to review the PR about this sandbox api and its preparations, so I split the original large PR into several smaller PRs. In fact, this PR was officially split from the PR https://github.com/kata-containers/kata-containers/pull/10330 two weeks ago, and this PR https://github.com/kata-containers/kata-containers/pull/10351 was also split from the previous big PR. You can see the split history here https://github.com/kata-containers/kata-containers/compare/ff6ba03c60b097fde432427b087a7d236261f6be..52397ca2c1a95e8fb9841e83f17d488c632cd362

**Comment by gkurz** [comment]
> Hi @gkurz
> 
> I understand your concerns, and as a community contributor, I also support your carefulness. Although this PR looks like a large diff, it is mainly code refactoring and location movement. Sorry, the commit message are indeed relatively brief. If you think it is necessary to modify the commit messages to be more detailed, I can revert this merge and add some detail info.
> 

The deed is done. I will certainly not ask to revert and push again. This is more a reminder for future contributions,
e.g. the final piece at #10356 .

> In addition, for this PR, I just want to make it easier for the reviewer to review the PR about this sandbox api and its preparations, so I split the original large PR into several smaller PRs. In fact, this PR was officially split from the PR #10330 two weeks ago, and this PR #10351 was also split from the previous big PR. You can see the split history here https://github.com/kata-containers/kata-containers/compare/ff6ba03c60b097fde432427b087a7d236261f6be..52397ca2c1a95e8fb9841e83f17d488c632cd362

Ok, I understand that you already did a substantial effort to split the changes and sometimes you don't gain much in splitting even more. Since it isn't obvious that the diff will be easy to review, I just suggest you wait some more before merging, even if you have the required approvals. One day is really too short.

Thanks @lifupan !

**Pull request merged**
